### PR TITLE
Cherry pick 34898 to sb3 branch

### PR DESCRIPTION
**Pull request by saragluna**
As title, to cherry pick #34898 to spring boot 3 branch.

### CODE REVIEW

File: sdk/spring/azure-spring-data-cosmos/pom.xml [hunk 1]
```diff
@@ -171,6 +181,18 @@
...
+      <artifactId>junit-vintage-engine</artifactId>
+      <version>5.9.1</version> <!-- {x-version-update;org.junit.vintage:junit-vintage-engine;external_dependency} -->
+      <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.hamcrest</groupId>
+          <artifactId>hamcrest-core</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-data-cosmos/pom.xml]: better use jupiter here instead of vintage engine.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -22,3 +22,10 @@
 
 /eng/versioning/external_dependencies.txt           @backwind1233 @chenrujun @hui1110 @netyyyy @saragluna @stliu @yiliuTo @moarychan @fangjian0423
 
+# PRLabel: %azure-spring
+/sdk/spring/spring-cloud-azure-appconfiguration-config*/ @mrm9084 @backwind1233 @chenrujun @hui1110 @netyyyy @saragluna @stliu @yiliuTo @moarychan @fangjian0423
+/sdk/spring/spring-cloud-azure-feature-management*/   @mrm9084 @backwind1233 @chenrujun @hui1110 @netyyyy @saragluna @stliu @yiliuTo @moarychan @fangjian0423
+/sdk/spring/spring-cloud-azure-starter-appconfiguration-config/ @mrm9084 @backwind1233 @chenrujun @hui1110 @netyyyy @saragluna @stliu @yiliuTo @moarychan @fangjian0423
+
+# PRLabel: %azure-spring
+/sdk/spring/azure-spring-data-cosmos/                @kushagraThapar @FabianMeiswinkel @backwind1233 @chenrujun @hui1110 @netyyyy @saragluna @stliu @yiliuTo @xinlian12 @moarychan @aayush3011 @simorenoh @fangjian0423 @jeet1995 @Pilchie
```
Comment (Netyyyy) [inline .github/CODEOWNERS:31]: Add a new line

File: sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/core/mapping/event/CosmosEventListenerUnitTests.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public void afterLoadEventGetsFilteredForDomainTypeWorksForSubtypes() {
         contactListener.onApplicationEvent(new AfterLoadEvent<>(NullNode.getInstance(), Person.class, "container-1"));
 
         assertThat(personListener.invokedOnAfterLoad).isTrue();
-        assertThat(contactListener.invokedOnAfterLoad).isTrue();
+        assertThat(contactListener.invokedOnAfterLoad).isFalse();
```
Comment (Netyyyy) [inline sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/core/mapping/event/CosmosEventListenerUnitTests.java:42]: Is that right for directly changing?

File: sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/integration/ContactRepositoryIT.java [hunk 4]
```diff
@@ -223,20 +223,20 @@ public void testShouldAllowListAndIterableResponses() {
...
-
-        List<Contact> contactsWithOffset = repository.getContactsWithOffsetLimit(1, 2);
-        Assert.assertEquals(2, contactsWithOffset.size());
-        Assert.assertEquals(TEST_CONTACT2, contactsWithOffset.get(0));
-        Assert.assertEquals(TEST_CONTACT3, contactsWithOffset.get(1));
-
-        List<ObjectNode> groupByContacts = repository.selectGroupBy();
-        Assert.assertEquals(3, groupByContacts.size());
-    }
+//    @Test
```
Comment (Netyyyy) [inline sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/integration/ContactRepositoryIT.java:226]: shall we need to add 'TODO'?

Comment (saragluna) [reply sdk/spring/azure-spring-data-cosmos/src/test/java/com/azure/spring/data/cosmos/repository/integration/ContactRepositoryIT.java:226]: This is a cherry pick, maybe we could leave the comment on #34898 directly.

File: sdk/spring/azure-spring-data-cosmos/src/test/resources/logback-test.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
   <root level="info">
     <appender-ref ref="STDOUT"/>
   </root>
-  <logger name="com.azure.cosmos" level="error"/>
+  <logger name="com.azure.cosmos" level="info"/>
   <logger name="org.springframework" level="error"/>
   <logger name="io.netty" level="error"/>
 </configuration>
```
Comment (Netyyyy) [inline sdk/spring/azure-spring-data-cosmos/src/test/resources/logback-test.xml:15]: add a new line

**Pull request merged**